### PR TITLE
Telegram bot support

### PR DESCRIPTION
**Pull request by kpaveliev**
### Description

Telegram bot support added. Now users can connect their telegram account to the app and:

- get list of their goals
- add new goals with title and category

### Additions to the app

- external packages: marshmallow_dataclass, marshmallow
- bot app with the all logic
- bot app: TgUser model to keep telegram users ids, chat ids, match app and telegram users
- bot app: TgUserUpdateView (/bot/verify) to verify users using front
- bot app: telegram API dataclasses to keep telegram responses
- bot app: TgClient to communicate with telegram API
- bot app: runbot manage.py command (key logic is here)
- goals app: Goal model updated to allow null values in due_date, description fields
- deploy: front version update
- deploy: additional container to communicate with the bot
- .env, github actions, github secrets: TG_TOKEN from the bot

### CODE REVIEW

File: todolist/bot/management/commands/runbot.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        reply = [f'#{goal.id} {goal.title}' for goal in goals]
+        return prefix + reply
+
+    def _create(self) -> list:
+        """Logic for /create command - list of categories"""
+        categories = Category.objects.filter(
+            board__participants__user=self.user.user,
+            board__participants__role__in=[BoardParticipant.Role.owner, BoardParticipant.Role.writer],
+            is_deleted=False
+        ).all()
```
Comment (painassasin) [inline todolist/bot/management/commands/runbot.py]: Думаю тут лучше использовать `only`, чтобы не тащить все поля модели, ведь тебе нужны только `id` и `title`

File: todolist/bot/management/commands/runbot.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        """Logic for /create command - create goal"""
+        self.goal = Goal.objects.create(
+            user=self.user.user, title=self.message, category=self.category
+        )
+        self.goal_mode = False
+        return f'Цель "{self.goal.title}" создана'
+
+    def _send_reply(self, reply: str | list) -> None:
+        """Send reply"""
+        if type(reply) == list:
```
Comment (painassasin) [inline todolist/bot/management/commands/runbot.py]: Типы лучше сравнивать через `isinstance`
```python
if isinstance(reply, list):
    ...
```

**Review by painassasin — APPROVED** [review]

File: todolist/bot/management/commands/runbot.py (hunk 1 above)
Comment (kpaveliev) [reply todolist/bot/management/commands/runbot.py]: Учтено

File: todolist/bot/management/commands/runbot.py (hunk 2 above)
Comment (kpaveliev) [reply todolist/bot/management/commands/runbot.py]: Учтено

**Pull request merged**